oop_student_report_project.py

```diff
@@ -0,0 +1,180 @@
+        self.p_marks = None
+        self.c_marks = None
+        self.eng_marks = None
+        self.percent = None
+        self.b_group = None
+        self.address = None
+        self.m_name = None
+        self.f_name = None
+        self.email = None
+
+
+     def add_student(self):
+        """
+        Prompts the user to input details for a student and calculates the overall percentage.
+        """
+
+        self.name = input("\nEnter the student's name: ")
+        self.roll_number = input("Enter the student's roll no: ")
+        self.m_marks = int(input("Enter the student's maths marks: "))
+        self.p_marks = int(input("Enter the student's physics marks: "))
+        self.c_marks = int(input("Enter the student's chemistry marks: "))
+        self.eng_marks = int(input("Enter the student's english marks: "))
+        self.percent = round(((self.m_marks + self.p_marks + self.c_marks + self.eng_marks)/5), 2)
+        self.b_group = input("Enter the student's blood group: ")
+        self.address = input("Enter the student's address: ")
+        self.m_name = input("Enter the student's mother's name: ")
+        self.f_name = input("Enter the student's father's name: ")
+        self.email = input("Enter the student's email: ")
+
+def add_info(class_info, student_count):
+    """
+    Add student information to the class details.
+
+    Args:
+    class_details (dict): Dictionary containing student information.
+    num_students (int): Current number of students.
+
+    Returns: (tuple: Updated class_details and num_students.)
+    """
+    student = Student()
+    student.add_student()
+    class_info[student.roll_number] = student
+    student_count += 1
+    return class_info,student_count
+
+def generate_report(total_students, class_data):
+    """
+    Generate a report based on student information.
+
+    Args:
+    num_students (int): Current number of students.
+    class_details (dict): Dictionary containing student information.
+
+    Returns: None
+    """
+    if  total_students == 0:
+        print("No student reports found in this classroom.")
+    else:
+        student_roll_no = input("Enter the roll no. of the student: ")
+        if student_roll_no in class_data:
+            concerned_student = class_data[student_roll_no]
+            report_table = PrettyTable(["Name","Roll no.","Maths","Physics","Chemistry","English","Total Percentage"])
+            report_table.add_row([concerned_student.name,concerned_student.roll_number,concerned_student.m_marks,concerned_student.p_marks,concerned_student.c_marks,concerned_student.eng_marks,concerned_student.percent])
+            print(report_table)
+        else:
+            print("No student data found,please enter a valid roll number.")
+
+def generate_identity(total_students, class_data):
+    """
+    Generate a student's ID.
+    
+    Args:
+    num_students (int): Current number of students.
+    class_details (dict): Dictionary containing student information.
```

Comment:
```suggestion
        num_students (int): Current number of students.
        class_details (dict): Dictionary containing student information.
```
Same here, the arguments are found no where, please rename.

Reply:
done